Sources/PackageLoading/ToolsVersionLoader.swift

```diff
@@ -88,30 +88,201 @@ extension Manifest {
+        public var isCompatibleWithPreSwift5_3_1: Bool {
+            leadingWhitespace.allSatisfy { $0 == "\n" } && toolsVersionSpecificationComponents.isCompatibleWithPreSwift5_3_1
+        }
+    }
+    
+    /// A representation of a Swift tools version specification in its constituent parts.
+    ///
+    /// A Swift tools version specification consists of the following parts:
+    ///
+    ///     //  swift-tools-version:  5.3.1
+    ///     ⌃~⌃~⌃~~~~~~~~~~~~~~~~~~~⌃~⌃~~~~
+    ///     │ │ └ label             │ └ version specifier
+    ///     │ └ spacing             └ spacing
+    ///     └ comment marker
+    ///
+    public struct ToolsVersionSpecificationComponents {
+        /// The comment marker.
+        ///
+        /// In a well-formed Swift tools version specification, the comment marker is `"//"`.
+        public let commentMarker: Substring
+        
+        /// The spacing after the comment marker
+        ///
+        /// In a well-formed Swift tools version specification, the spacing after the comment marker is a continuous sequence of horizontal whitespace characters.
+        ///
+        /// For Swift ≤ 5.3, the spacing after the comment marker must be a single `U+0020`.
+        public let spacingAfterCommentMarker: Substring
+        
+        /// The label part of the Swift tools version specification.
+        ///
+        /// In a well-formed Swift tools version specification, the label is `"swift-tools-version:"`
+        public let label: Substring
+        
+        /// The spacing between the label part of the Swift tools version specification and the version specifier.
+        ///
+        /// In a well-formed Swift tools version specification, the spacing after the label is a continuous sequence of horizontal whitespace characters.
+        ///
+        /// For Swift ≤ 5.3, no spacing is allowed after the label.
+        public let spacingAfterLabel: Substring
+        
+        /// The version specifier.
+        public let versionSpecifier: Substring
+        
+        /// A Boolean value indicating whether everything up to the version specifier in the Swift tools version specification represented in its constituent parts is well-formed.
+        public var everythingUpToVersionSpecifierIsWellFormed: Bool {
+            // FIXME: Make `label` case sensitive?
+            // FIXME: Replace with `commentMarker == "//" && (label == "swift-tools-version:" || label.lowercased() == "swift-tools-version:")`?
+            // "swift-tools-version:" has more than 15 UTF-8 code units, so `label` is likely to have more than 15 UTF-8 code units too.
+            // Strings with more than 15 UTF-8 code units are heap-allocated on 64-bit platforms, 10 on 32-bit platforms.
+            // `lowercase()` returns a heap-allocated string here, and this is inefficient, although the inefficiency is perhaps insignificant.
```

Comment:
This allocation is most likely insignificant, since this code should only be called once per manifest, IIUC.

Reply:
Yep it is rather insignificant. I put a FIXIT there, partially because of the case-insensitivity in "swift-tools-version", and this is just a "con" of the case-insensitivity.

Reply:
Understood.  I think it's good to call out but not something to worry about for the PR.

Reply:
In 70ce4c36a9a0a3cbf4a9e8b83c592d067f630faf, I removed the `FIXME` that suggests to make the label case-sensitive. I kept the few lines regarding the `String` allocation as an argument for case-sensitivity, but reworded it a bit, and made it clear that the inefficiency in the potential heap allocation is insignificant.

Reply:
Thanks!